keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -1065,6 +1107,9 @@ Recall that end users cannot usually observe component logs or access metrics.
 - [ ] Other (treat as last resort)
   - Details:
 
+TODO(haircommander): what's the best way to check this? If the kubelet ignores the field in the pod spec, then KAS could request a
+pod be created with an image volume and report that intent, but the kubelet could have not actually mounted it.
```

Comment:
what does an old kubelet do?  I would expect the unknown image type to result in the pod being stuck in a pending phase forever.

Reply:
it looks like the pod is created without the volume https://github.com/kubernetes/kubernetes/blob/815532749c551413f8b9d9be0ae1f24cb3a1e871/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L1383

Reply:
> it looks like the pod is created without the volume https://github.com/kubernetes/kubernetes/blob/815532749c551413f8b9d9be0ae1f24cb3a1e871/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L1383

Ok, I can probably live with that too, but this section needs to indicate some helpful bash hints for knowing whether the volume was indeed mounted.

Reply:
What is the desired state for the created pod when the feature-gate is disabled on kubelet or the pod is scheduled on a node with older kubelet version?

Is the pod running or failed?

From kubelet's volume manager perspective, it will fail to mount the volume into pod because the image volume plugin is not registered in the kubelet when the feature-gate is disabled or the pod is scheduled on a node with older kubelet version. It leads to the pod failed to start. The pod will be in the Pending state.

- https://github.com/kubernetes/kubernetes/blob/06fb818f86e04770fb25b974d9c716bd519e093d/pkg/kubelet/volumemanager/volume_manager.go#L393
- https://github.com/kubernetes/kubernetes/blob/06fb818f86e04770fb25b974d9c716bd519e093d/pkg/volume/util/util.go#L534
- https://github.com/kubernetes/kubernetes/blob/06fb818f86e04770fb25b974d9c716bd519e093d/pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go#L305

Reply:
ah that's even better IMO. I'll update it to reflect that, I misunderstood the code